tests/0143-exponential_backoff.c

```diff
@@ -0,0 +1,571 @@
+                    RD_KAFKAP_FindCoordinator) {
+                        if (previous_request_ts != -1) {
+                                int64_t time_difference =
+                                    (rd_kafka_mock_request_timestamp(
+                                         requests[i]) -
+                                     previous_request_ts) /
+                                    1000;
+                                TEST_ASSERT(((time_difference > low - buffer) &&
+                                             (time_difference < low + buffer)),
+                                            "Time difference should be close "
+                                            "to 1 second, it is %" PRId64
+                                            " ms instead.\n",
+                                            time_difference);
+                                retry_count++;
+                        }
+                        previous_request_ts =
+                            rd_kafka_mock_request_timestamp(requests[i]);
+                }
+        }
+        rd_kafka_destroy(consumer);
+        for (i = 0; i < request_cnt; i++)
+                rd_kafka_mock_request_destroy(requests[i]);
+        rd_free(requests);
+        rd_kafka_mock_clear_requests(mcluster);
+        SUB_TEST_PASS();
+}
+
+/**
+ * Exponential Backoff needs to be checked for the request_type. Also the
+ * request_type should only be retried if one previous has failed for correct
+ * execution.
+ */
+int helper_exponential_backoff(rd_kafka_mock_cluster_t *mcluster,
+                               int32_t request_type) {
+        rd_kafka_mock_request_t **requests = NULL;
+        size_t request_cnt                 = 0;
+        int64_t previous_request_ts        = -1;
+        int32_t retry_count                = 0;
+        int32_t error_code                 = 0;
+        size_t i;
+        requests = rd_kafka_mock_get_requests(mcluster, &request_cnt);
+        for (i = 0; i < request_cnt; i++) {
+                if (rd_kafka_mock_request_api_key(requests[i]) ==
+                    request_type) {
+                        TEST_SAY(
+                            "Broker Id : %d API Key : %d Timestamp : %" PRId64
+                            "\n",
+                            rd_kafka_mock_request_id(requests[i]),
+                            rd_kafka_mock_request_api_key(requests[i]),
+                            rd_kafka_mock_request_timestamp(requests[i]));
+                        if (previous_request_ts != -1) {
+                                int64_t time_difference =
+                                    (rd_kafka_mock_request_timestamp(
+                                         requests[i]) -
+                                     previous_request_ts) /
+                                    1000;
+                                int64_t low =
+                                    ((1 << retry_count) * (retry_ms)*75) / 100;
```

Comment:
Add comment above saying you are using 25% buffer on each side, and why.

Reply:
Not resolved.

Reply:
Have made the comment given jitter percent is 20 so to account for latency we have the buffer set to 25.